internal/controllers/topology/cluster/reconcile_state.go

```diff
@@ -517,6 +560,23 @@ func (r *Reconciler) createMachineDeployment(ctx context.Context, s *scope.Scope
 	}
 	r.recorder.Eventf(cluster, corev1.EventTypeNormal, createEventReason, "Created %q", tlog.KObj{Obj: md.Object})
 
+	// Wait until MachineDeployment is visible in the cache.
+	// Note: We have to do this because otherwise using a cached client in current state could
+	// miss a newly created MachineDeployment (because the cache might be stale).
+	err = wait.PollUntilContextTimeout(ctx, 5*time.Millisecond, 1*time.Second, true, func(ctx context.Context) (bool, error) {
```

Comment:
```suggestion
	err = wait.PollUntilContextTimeout(ctx, 50*time.Millisecond, 5*time.Second, true, func(ctx context.Context) (bool, error) {
```
I would give some time more, so we are more resilient in case of temporary network issue/slowness, also no need to be too aggressive to the API server for an operation that doesn't happe frequently. Same for delete

Reply:
We extend the time we wait, but this is just calling the local cache. This won't lead to any calls to the apiserver